src/main/java/com/rultor/agents/IndexesRequests.java

```diff
@@ -39,8 +45,104 @@
  * @version $Id$
  */
 public final class IndexesRequests implements SuperAgent {
+    /**
+     * Xpath expression for retrieving log entries from a talk's XML.
+     */
+    private static final String ARCHIVE_LOG = "//archive/log";
+
+    /**
+     * Name of the log index attribute.
+     */
+    private static final String INDEX = "index";
+
     @Override
+    @SuppressWarnings("PMD.AvoidInstantiatingObjectsInLoops")
     public void execute(final Talks talks) throws IOException {
-        throw new UnsupportedOperationException();
+        int maxTalkIndex = this.getMaxTalkIndex(talks);
```

Comment:
why can't we just name this `max`? check this: http://www.yegor256.com/2014/04/27/typical-mistakes-in-java-code.html#variable-names